File: Problem_Sets/LeetCode/TwoSums/TwoSum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        dict = {value: key for key, value in enumerate(nums)}
        for i, num in enumerate(nums):
            complement = target - num
            if complement in dict and dict[complement] != i:
                return [dict[complement], i]

        return []

File: Problem_Sets/LeetCode/TwoSums/TwoSum.cpp
#include <unordered_map>

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        // O(n)
        unordered_map<int, int> hashtable;

        // Store current array value in hash table
        for (int i = 0; i < nums.size(); ++i) { 
            hashtable[nums[i]] = i;
        }

        for (int i = 0; i < nums.size(); ++i) {
            int complement = target - nums[i];
             if (hashtable.find(complement) != hashtable.end() && hashtable[complement] != i) {
                return {hashtable[complement], i};
            }
        }

        return {};
    }
};